tests/functional/test_eclab_parser.py

```diff
@@ -0,0 +1,406 @@
+        "cycle number": ("", (1.000000000000000e000, 2.000000000000000e000)),
+        "(Q-Qo)/C": ("C", (0.0000000e000, 7.0078465e-007)),
+        "I Range": ("", (40, 40)),
+        "<Ece>/V": ("V", (1.0487431e000, 1.0656431e000)),
+        "P/W": ("W", (3.1451185e-007, 5.5134848e-007)),
+        "Ewe-Ece/V": ("V", (-1.9443744e-001, -2.1177679e-001)),
+        "loop_number": ("", (0, 1)),  # added by Ixdat
+        "Ns": ("", (0, 0)),  # added by Ixdat
+    },
+    "dataset_with_loop/dataset_with_loop_01_03_CP_DUSB0_C01.mpt": {
+        "mode": ("", (1, 1)),
+        "ox/red": ("red", (1, 1)),
+        "error": ("", (0, 0)),
+        "control changes": ("", (0, 0)),
+        "Ns changes": ("", (0, 0)),
+        "counter inc.": ("", (0, 0)),
+        "Ns": ("", (0, 0)),
+        "time/s": ("s", (5.109679870918626e001, 1.122481971643720e002)),
+        "control/mA": ("mA", (0.0000000e000, 0.0000000e000)),
+        "Ewe/V": ("V", (8.1625509e-001, 8.4291708e-001)),
+        "I/mA": ("mA", (2.7354923e-005, 3.2436059e-005)),
+        "dQ/C": ("C", (0.0000000e000, 3.2346591e-007)),
+        "(Q-Qo)/C": ("C", (0.0000000e000, 3.2346591e-007)),
+        "half cycle": ("", (0, 0)),
+        "Q charge/discharge/mA.h": (
+            "mA.h",
+            (0.000000000000000e000, 8.985164472556385e-008),
+        ),
+        "I Range": ("", (41, 41)),
+        "Ece/V": ("V", (1.0758879e000, 1.0688440e000)),
+        "Q discharge/mA.h": ("mA.h", (0.000000000000000e000, 0.000000000000000e000)),
+        "Q charge/mA.h": ("mA.h", (0.000000000000000e000, 8.985164472556385e-008)),
+        "Capacity/mA.h": ("mA.h", (0.000000000000000e000, 8.985164472556385e-008)),
+        "Efficiency/%": ("%", (0.0000000e000, 0.0000000e000)),
+        "cycle number": ("", (0.000000000000000e000, 0.000000000000000e000)),
+        "P/W": ("W", (2.2328596e-008, 2.7340908e-008)),
+        "Ewe-Ece/V": ("V", (-2.5963283e-001, -2.2592688e-001)),
+        "loop_number": ("", (0, 1)),  # added by Ixdat
+    },
+}
+
+
+@fixture(scope="module", params=tuple(TEST_DATA.items()))
+@mark.external
+def measurements_with_data(request):
+    """Load all measurements from files and connect it with the test data."""
+
+    filename = request.param[0]
+    column_data = request.param[1]
+    measurement = Measurement.read(DATA_DIR / filename, reader="biologic")
+    return measurement, column_data
+
+
+@mark.external
+def test_shape(measurements_with_data):
+    """Test the shape of the parsed data from the measurements.
+
+    The method is testing whether the amount and the names
+    of the columns are the same.
+
+    """
+    measurement = measurements_with_data[0]
+    column_names = measurements_with_data[1].keys()
+
+    # test amount of columns
+    assert len(measurement.series_list) == len(column_names)
+
+    # test names of columns
+    for name in column_names:
+        if name not in measurement.series_names:
+            assert (
```

Comment:
If this is for a clearer error message then great (although I think the error message may be a copy/paste from another test, at least it isn't very informative for this test). Otherwise you can just do:
```python
assert name in measurement.series_name
```
and pytest will show you the value it fails for if it does, which, along with an aptly named test should contain all the information you would need.